config/confighttp/confighttp.go

```diff
@@ -410,6 +411,14 @@ func WithDecoder(key string, dec func(body io.ReadCloser) (io.ReadCloser, error)
 	})
 }
 
+// WithSpanPrefix creates a span prefixed with the specified name.
+// Ideally, this prefix should be the component's ID.
+func WithSpanPrefix(spanPrefix string) ToServerOption {
+	return toServerOptionFunc(func(opts *toServerOptions) {
+		opts.spanPrefix = spanPrefix
+	})
+}
```

Comment:
We should make it a bit more generic and accept `otelhttp.Option` or at least accept exactly same formatter? You may now want this, but later people may want other things, etc.

Reply:
The problem though, is that otelhttp is not stable, and we are looking to mark this stable soon. So I think we may not be able to accept this until `otelhttp` is marked stable.

Reply:
@dmathieu since you are around, any idea of whether `otelhttp` will be marked stable any time soon?

Reply:
@sfc-gh-bdrutu @bogdandrutu can you take a look now? I've made it more generic.

Reply:
> any idea of whether otelhttp will be marked stable any time soon?

That is not on our radar at the moment. However, we are committed to keeping the API stable, or providing deprecation warnings for at least two versions.

Reply:
@sfc-gh-bdrutu @bogdandrutu can you take a look?

Reply:
A possible way around this is to move the option to a separate module. We are discussing this in #11769